### DISCUSSION TITLE
Adding separate textures under BSDF

### DISCUSSION THREAD

**Discussion by baher3d** [mitsuba-renderer/mitsuba3#617] [Q&A] [answered]
I am trying to render a scene with a 3D human object. My materials are a diffuse map, a normals map, and multiple specular textures of different object parts. How can I import it into my scene? The only BSDF similar to what I need is [The-Principled-Bsdf](`https://mitsuba.readthedocs.io/en/stable/src/generated/plugins_bsdfs.html#the-principled-bsdf-principled`) but it doesn't have a normals option and the option of adding multiple specular textures. 

What I'm currently using is:
```
'material':
        'type': 'diffuse',
        'reflectance': {
            'type': 'bitmap',
            'filename': 'diff_texture.jpg',
            'wrap_mode': 'mirror'
        }
```

**Comment by njroussel** [comment, marked as answer]
Hi @baher3d 

The `normalmap` plugin can take a nested BSDF as you can see in [the documentation](https://mitsuba.readthedocs.io/en/stable/src/generated/plugins_bsdfs.html#normal-map-bsdf-normalmap). So, you can combine it with any BSDF really.

From your description, I'd recommend the `prinipled` BSDF too. I'm unsure why you have multiple specular textures? Can you not just merge them into a single one (your UVs shouldn't be overlapping)? Or split your object itself eventually?
You might even want to take a look at the [`blendbsdf`](https://mitsuba.readthedocs.io/en/stable/src/generated/plugins_bsdfs.html#blended-material-blendbsdf). If its easier to create separate materials and then merge them for your shape. It would require a texture to map your "different object parts".

**Comment by baher3d** [comment]
Hey @njroussel Thanks a lot for the fast response! The objects dataset was bought online, I'm also unsure about the logic of separating them. I think the best option is as you recommended, I will try to merge them and use the principled BSDF.

Thanks again!
